packages/maskbook/src/plugins/NFT/UI/NFTInPost.tsx

```diff
@@ -0,0 +1,43 @@
+import { useAsync } from 'react-use'
+import Services from '../../../extension/service'
+import MaskbookPluginWrapper from '../../MaskbookPluginWrapper'
+import type { NFTInPostProps, TokenDetails } from '../types'
+import { parseNftUrl } from '../utils'
+import NFTCardUI from './NFTCardUI'
+import { openseaAssetsApiUrl } from '../constants'
+
+export default function NFTInPost(props: NFTInPostProps) {
+    const tokenDetailsResponse = useAsync(
+        async function () {
+            let updatedTokenDetails: TokenDetails = {}
+            const [address, tokenId] = parseNftUrl(props.nftUrl)
+
+            // if nftUrl is invalid
+            if (!(address && tokenId)) return updatedTokenDetails
+
+            const link = new URL(openseaAssetsApiUrl)
+            link.searchParams.set('asset_contract_address', address)
+            link.searchParams.set('token_ids', tokenId)
+
+            const response = await Services.Helper.fetch(link.toString())
+            const { assets } = JSON.parse(await response.text())
+
+            // if asset is not found
+            if (assets.length === 0) return updatedTokenDetails
+
+            if (assets[0].name) updatedTokenDetails.name = assets[0].name
```

Comment:
What will happen if it has multiple assets?

Reply:
that isn't handled here, but I was thinking of a group of images for collections and multiple assets.